test/load-generator/boulder-calls.go

```diff
@@ -168,7 +168,7 @@ func newOrder(s *State, c *acmeCache) error {
 	// Generate that many random domain names. There may be some duplicates, we
 	// don't care. The ACME server will collapse those down for us, how handy!
 	dnsNames := []identifier.ACMEIdentifier{}
-	for i := 0; i <= orderSize; i++ {
+	for range orderSize {
```

Comment:
This is off-by-one. Assuming that orderSize on both sides is 3
```
func main() {
	orderSize := 3
	for i := 0; i <= orderSize; i++ {
		fmt.Printf("old %d\n", i)
	}
	for i := range orderSize {
		fmt.Printf("new %d\n", i)
	}
}
```
```
old 0
old 1
old 2
old 3
new 0
new 1
new 2
```

Reply:
Correct. But this is fixing an off-by-one error: previously if "orderSize" was 3, it would generate 4 dnsNames. Now when orderSize is 3 is actually generates 3.